plugins/firehydrant/src/components/hooks.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Entity } from '@backstage/catalog-model';
+
+export const FIREHYDRANT_SERVICE_NAME_ANNOTATION =
+  'firehydrant.com/service-name';
+/** @public */
+export const isFireHydrantAvailable = (entity: Entity) =>
+  Boolean(entity.metadata.annotations?.[FIREHYDRANT_SERVICE_NAME_ANNOTATION]);
+export const getFireHydrantServiceName = (entity: Entity) =>
+  isFireHydrantAvailable(entity)
+    ? entity?.metadata.annotations?.[FIREHYDRANT_SERVICE_NAME_ANNOTATION] ?? ''
+    : `${entity?.kind}:${entity?.metadata?.namespace ?? 'default'}/${
```

Comment:
We typically use `stringifyEntityRef` from `@backstage/catalog-model`. That one does just this, with the important caveat that it lowercases everything. However, that's the exact form that we always want to use for cross-system correlation of references to entities - it's essentially to be regarded as an opaque, unique identifier. It's what all other plugins do. Is there some particuar reason that you want to preserve the casing here, or could you stringify it in the standard way?

Reply:
Interesting - good to know about `stringifyEntityRef`

All I can say is that's how the plugin was [originally written](https://github.com/backstage/backstage/blob/master/plugins/firehydrant/src/components/ServiceDetailsCard/ServiceDetailsCard.tsx#L154). 🤷🏼 
I don't know if it is (or was) a limitation of the FireHydrant API, but since I don't know _why_ it was like that, I didn't want to break existing functionality by changing it. Chesterton's fence and all.

Thoughts?

Reply:
A quick test of the FireHydrant API suggests that it does indeed work with an all lowercase version of the string.
I can swap it to `stringifyEntityRef`